src/kubernetes/client/config/in_cluster.rs

```diff
@@ -0,0 +1,68 @@
+//! Everything related to building in-cluster configuration.
+
+use super::Config;
+use crate::tls::TlsOptions;
+use http02::Uri;
+use snafu::{ResultExt, Snafu};
+
+impl Config {
+    /// Prepares a config sutable for use when running in k8s cluster.
+    pub fn in_cluster() -> Result<Self, Error> {
+        let host = std::env::var("KUBERNETES_SERVICE_HOST").context(NotInCluster)?;
+        let port = std::env::var("KUBERNETES_SERVICE_PORT").context(NotInCluster)?;
+
+        let base = Uri::builder()
+            .scheme("https")
+            .authority(join_host_port(host.as_str(), port.as_str()).as_str())
+            .path_and_query("/")
+            .build()
+            .context(InvalidUrl)?;
+
+        let token_file = "/var/run/secrets/kubernetes.io/serviceaccount/token";
+        let root_ca_file = "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt";
+
+        let token = std::fs::read_to_string(token_file).context(Token)?;
+
+        let mut tls_options = TlsOptions::default();
+        tls_options.ca_path = Some(root_ca_file.into());
+
+        Ok(Self {
+            base,
+            token,
+            tls_options,
+        })
+    }
+}
+
+/// An error returned when building an in-cluster configuration.
+#[derive(Debug, Snafu)]
+pub enum Error {
+    /// The in-cluster configuration requested while executing not in a cluster
+    /// environment.
+    #[snafu(display("unable to load in-cluster configuration, KUBERNETES_SERVICE_HOST and KUBERNETES_SERVICE_PORT must be defined"))]
```

Comment:
Can we hint which one is not set through a `missing` field?

Reply:
We can actually! I copied the error almost verbatim from the k8s code, but we can do better.

Reply:
I'm surprised there isn't a rust client that can do this stuff for us?

Reply:
There is, and not just one, but a couple. They isn't one that would be a good fit for us though.

A lot of things in this PR are actually based on the approach taken by the previous k8s implementation, to reuse code and design decisions where possible.
I assure you that I double-checked the options out there before taking the path of building a custom client. But this is what was decided on the last time we built this stuff, and this is still the way to go.

Reply:
Ok. :) Fair. I looked around and came to the same conclusion.